NameLayerMain/src/vg/civcraft/mc/namelayer/gui/MainGroupGUI.java

```diff
@@ -0,0 +1,1272 @@
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g.getAllMembers(PlayerType.MODS)
+									.size()) + " mods");
+		}
+		if (gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("ADMINS"))
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g.getAllMembers(PlayerType.ADMINS)
+									.size()) + " admins");
+		}
+		if (gm.hasAccess(g, p.getUniqueId(),
+				PermissionType.getPermission("OWNER"))
+				|| hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.AQUA
+							+ String.valueOf(g.getAllMembers(PlayerType.OWNER)
+									.size()) + " owner");
+		}
+		if (hasGroupStatsPerm) {
+			ISUtils.addLore(
+					is,
+					ChatColor.DARK_AQUA
+							+ String.valueOf(g.getAllMembers().size())
+							+ " total group members");
+			ISUtils.addLore(is, ChatColor.DARK_AQUA + "Group owner: "
+					+ ChatColor.YELLOW + NameAPI.getCurrentName(g.getOwner()));
+		}
+		c = new DecorationStack(is);
+		return c;
+	}
+
+	/**
+	 * Utility to get a "nice" version of the rank the given player has in the
+	 * group this gui is based on
+	 * 
+	 * @param uuid
+	 *            Player whose rank name should be checked
+	 * @return Rankname of the given player ready to be put into the gui or null
+	 *         if the given UUID was null or the player didn't have an explicit
+	 *         rank in the group
+	 */
+	private String getRankName(UUID uuid) {
+		if (uuid == null) {
+			return null;
+		}
+		PlayerType pType = g.getPlayerType(uuid);
+		if (pType == null) {
+			return null;
+		}
+		return getDirectRankName(pType);
+	}
+
+	public static String getDirectRankName(PlayerType pType) {
+		String res = pType.toString().toLowerCase();
+		char[] chars = res.toCharArray();
+		chars[0] = new String(new char[] { chars[0] }).toUpperCase()
+				.toCharArray()[0];
+		res = new String(chars);
+		if (res.endsWith("s")) {
+			return res.substring(0, res.length() - 1);
+		}
+		return res;
+	}
+
+	/**
+	 * Gets the permission needed to make changes to the given PlayerType
+	 * 
+	 * @param pt
+	 *            PlayerType to get permission for
+	 * @return Permission belonging to the given PlayerType or null if the
+	 *         permission was NOT_BLACKLISTED or null
+	 */
+	public static PermissionType getAccordingPermission(PlayerType pt) {
+		switch (pt) {
+		case MEMBERS:
+			return PermissionType.getPermission("MEMBERS");
+		case MODS:
+			return PermissionType.getPermission("MODS");
+		case ADMINS:
+			return PermissionType.getPermission("ADMINS");
+		case OWNER:
+			return PermissionType.getPermission("OWNER");
+		}
+		return null;
+	}
+
+	/**
+	 * Utility to determine whether the player is being promoted or demoted
+	 */
+	private static String demoteOrPromote(PlayerType oldRank,
+			PlayerType newRank, boolean upperCaseFirstLetter) {
+		int oldIndex = 0;
+		int newIndex = 0;
+		for (int i = 0; i < PlayerType.values().length; i++) {
```

Comment:
I might be forgetting but aren't there utility functions now on PlayerType that allows direct conversion of type to "ordinal rank" -- so the loop here could probably be removed and replaced with those switch-case based conversions.

Even better might be if PlayerType just had "relative ordinal rank" as a parameter with an accessor, or something along those lines (static compare perhaps?)